scrapers/il/bills.py

```diff
@@ -581,56 +579,54 @@ def scrape_documents(self, bill, version_url):
-                        doctype = re.search(
-                            r"DocTypeId=(.*?)&", url, flags=re.IGNORECASE
-                        ).group(1)
-                        # numeric component of the session id
-                        session_number = int(
-                            "".join(
-                                char
-                                for char in bill.legislative_session
-                                if char.isdigit()
-                            )
-                        )
-
-                        # if it's html, extract the pdf link too while we're here.
-                        pdf_url = f"https://ilga.gov/legislation/{session_number}/{doctype}/PDF/{version_id}.pdf"
-                        bill.add_version_link(
-                            name, pdf_url, media_type="application/pdf"
-                        )
-
-                    bill.add_version_link(name, url, media_type=mimetype)
-                elif name in FULLTEXT_DOCUMENT_TYPES:
-                    bill.add_document_link(name, url)
-                elif "Printer-Friendly" in name:
-                    pass
+            if name in VERSION_TYPES or "amendment" in name.lower():
+                if pdf_only:
+                    # eed to visit the version's page, and get PDF link from there
+                    # otherwise get a faulty "latest version"/"LV" alias/duplicate
+                    url = "{}&Print=1".format(url)
+                    version_page_html = self.get(url).text
+                    version_page_doc = lxml.html.fromstring(version_page_html)
+                    version_page_doc.make_links_absolute(url)
+                    pdf_link = version_page_doc.xpath('//a[contains(@href, "PDF")]')
+                    if not pdf_link:
+                        continue
+                    pdf_link = pdf_link[0]
+                    url = pdf_link.get("href")
+                    mimetype = "application/pdf"
                 else:
-                    self.warning("unknown document type %s - adding as document" % name)
-                    bill.add_document_link(name, url)
+                    url = "{}&Print=1".format(url)
+                    mimetype = "text/html"
+                    version_id = re.search(
+                        r"DocName=(.*?)&", url, flags=re.IGNORECASE
+                    ).group(1)
+                    doctype = re.search(
+                        r"DocTypeId=(.*?)&", url, flags=re.IGNORECASE
+                    ).group(1)
+                    # numeric component of the session id
+                    session_number = int(
+                        "".join(
+                            char for char in bill.legislative_session if char.isdigit()
+                        )
+                    )
+                    # if it's html, extract the pdf link too while we're here.
+                    pdf_url = f"https://beta.ilga.gov/documents/legislation/{session_number}/{doctype}/PDF/{version_id}.pdf"
```

Comment:
Please use a `BASE_URL` constant (or similar variable) in place of the literal string `https://beta.ilga.gov`

Reply:
Fixed issue